src/pages/workspaces/workspace/delete-workspace-modal.state.ts

```diff
@@ -0,0 +1,111 @@
+import _ from 'lodash/fp'
+import { useState } from 'react'
+import { Ajax } from 'src/libs/ajax'
+import { reportError, withErrorReportingInModal } from 'src/libs/error'
+import { useCancellation, useOnMount } from 'src/libs/react-utils'
+import { getUser } from 'src/libs/state'
+import * as Utils from 'src/libs/utils'
+import { BaseWorkspace, isGoogleWorkspace, WorkspaceInfo } from 'src/libs/workspace-utils'
+import { isResourceDeletable } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+
+
+interface DeleteWorkspaceModalLeoApp {
+  appName: string
+  status: string
+  cloudContext: any
+}
+
+export interface DeleteWorkspaceState {
+  loading: boolean
+  deleting: boolean
+  isDeleteDisabledFromResources: boolean
+  workspaceBucketUsageInBytes: number | null
+  deletableApps: DeleteWorkspaceModalLeoApp[]
+  nonDeletableApps: DeleteWorkspaceModalLeoApp[]
+  collaboratorEmails: string[] | null
+  hasApps: () => boolean
+  deleteWorkspace: () => void
+  controlledResourcesExist: boolean
+}
+
+
+export const useDeleteWorkspaceState = (workspace: BaseWorkspace, onDismiss: () => void, onSuccess: () => void) : DeleteWorkspaceState => {
```

Comment:
consider moving these hook arguments into a single hookArgs object parameter with a DeleteWorkspaceStateArgs interface defining their names and types.  Argument bags have a lot of maintenance and readability advantages for high-profile interface surfaces (i.e. a visual component's props).